src/formatter-settings/index.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+export class JavaFormatterSettingsEditorProvider implements vscode.CustomTextEditorProvider {
+
+    public static readonly viewType = "java.formatterSettingsEditor";
+
+    constructor(private readonly context: vscode.ExtensionContext) {
+    }
+
+    public async showFormatterSettingsEditor() {
+        // Use a fake profile
+        const defaultProfile: string = path.join(this.context.extensionPath, "webview-resources", "formatter.xml");
+        const resource = vscode.Uri.file(defaultProfile);
+        vscode.commands.executeCommand("vscode.openWith", resource, "java.formatterSettingsEditor");
+    }
+
+    public async resolveCustomTextEditor(_document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel, _token: vscode.CancellationToken): Promise<void> {
```

Comment:
If the user open a pom.xml via this custom editor, what will happen? At least we need to have some message telling users that it's only supported for the format xml file

Reply:
He will get an error telling him this file doesn't contain any valid setting, and we'll have these logics in the next PR.